### ISSUE TITLE
ClassAndModuleChildren incorrectly flags classes with a single method

### ISSUE THREAD

**Issue by CyborgMaster** [rubocop/rubocop#1094]
I like the compact style of class and method definition and therefore have the following in my `.rubocop.yml`:

``` yaml
ClassAndModuleChildren:
  EnforcedStyle: compact
```

However this erroneously flags the following module:

``` ruby
module TimeCopMinitestPlugin
  def after_teardown
    Timecop.return
    super
  end
end
```

with this offense message:

```
test/test_helper.rb:19:8: C: Use compact module/class definition instead of nested style.
module TimeCopMinitestPlugin
       ^^^^^^^^^^^^^^^^^^^^^
```

It seems as if it is detecting that the class has a single child and therefore prompting me to try and combine it.  This works for classes/modules with a single inner class/module, but is impossible when the contents is a method.  

The offense should not be reported if the single child is not a class/module.
